model.py

```diff
@@ -2272,18 +2273,33 @@ def author_contributors(self):
                 l = acceptable_substitutes[x.role]
                 if x.contributor not in l:
                     l.append(x.contributor)
+
+        def dedupe(l):
+            """If an item shows up multiple times in a list, 
+            keep only the first occurance.
+            """
+            seen = set()
+            deduped = []
+            for i in l:
+                if i in seen:
+                    continue
+                deduped.append(i)
+                seen.add(i)
+            return deduped
+
         if primary_author:
-            return [primary_author] + sorted(other_authors, key=lambda x: x.name)
+            return dedupe([primary_author] + sorted(other_authors, key=lambda x: x.name))
```

Comment:
In this case we're trying to figure out what would go on the book cover. The primary author should go first, and be followed by non-primary authors in alphabetical order. People whose role does not rise to the level of "authorship" (e.g. illustrator) do not show up. 

The list as a whole should contain no duplicates. This might happen because someone is erroneously listed twice in the same role, someone is listed as both primary author and regular author, someone is listed as both author and translator, etc. However it happens, your name only shows up once on the front of the book.